### PR TITLE
Distributed caches are mem caches

### PR DESCRIPTION
**Pull request by ChristophWurst**
Memcached and Redis are instances of IMemcache. That interface is
superior as it has the fancy inc and dec helpers.

For app development it's nice if we can use memcache without an instanceof check.

### CODE REVIEW

**Comment by come-nc** [comment]
Lots of tests needs to be adapted it seems. I do not understand the psalm error: "Type OCP\IMemcache for $lockingCache is always OCP\IMemcache".
Where do me see the file:line information for psalm errors from the CI?

File: apps/files_external/tests/Service/StoragesServiceTest.php [hunk 1]
```diff
@@ -315,7 +318,8 @@ public function testDeleteStorage($backendOptions, $rustyStorageId) {
 		$mountCache->clear();
 		$user = $this->createMock(IUser::class);
 		$user->method('getUID')->willReturn('test');
-		$cache = $this->createMock(ICache::class);
+		$cache = $this->cache = $this->createMock(IMemcache::class);
+		;
```
Comment (come-nc) [inline apps/files_external/tests/Service/StoragesServiceTest.php:322]:
```suggestion
		$cache = $this->cache = $this->createMock(IMemcache::class);
```

File: apps/theming/tests/ThemingDefaultsTest.php [hunk 2]
```diff
@@ -80,7 +85,8 @@ protected function setUp(): void {
 		$this->l10n = $this->createMock(IL10N::class);
 		$this->urlGenerator = $this->createMock(IURLGenerator::class);
 		$this->cacheFactory = $this->createMock(ICacheFactory::class);
-		$this->cache = $this->createMock(ICache::class);
+		$this->cache = $this->cache = $this->createMock(IMemcache::class);
+		;
```
Comment (come-nc) [inline apps/theming/tests/ThemingDefaultsTest.php:89]:
```suggestion
		$this->cache = $this->createMock(IMemcache::class);
```

File: apps/files_trashbin/tests/StorageTest.php [hunk 3]
```diff
@@ -561,7 +561,7 @@ public function testSingleStorageDeleteFolderFail() {
 	 */
 	public function testShouldMoveToTrash($mountPoint, $path, $userExists, $appDisablesTrash, $expected) {
 		$fileID = 1;
-		$cache = $this->createMock(ICache::class);
+		$cache = $this->cache = $this->createMock(IMemcache::class);
```
Comment (come-nc) [inline apps/files_trashbin/tests/StorageTest.php:564]:
```suggestion
		$cache = $this->createMock(IMemcache::class);
```

**Review by come-nc — CHANGES_REQUESTED** [review]
Seems like a case of batch replace gone wrong.
Lots of superflous "$this->cache =" and duplicated ";".

**Comment by ChristophWurst** [comment]
I've raised https://github.com/nextcloud/server/issues/31787 so this can be picked up later.

**Pull request closed without merging**